src/Sarif/FileRegionsCache.cs

```diff
@@ -184,17 +186,15 @@ public Region ConstructMultilineContextSnippet(Region inputRegion, Uri uri, stri
             region.EndColumn = 0;
             region.StartLine = 0;
             region.EndLine = 0;
-            region.CharOffset = originalRegion.CharOffset < smallSnippetLength
-                ? 0
-                : originalRegion.CharOffset - smallSnippetLength;
+            region.CharOffset = Math.Max(0, originalRegion.CharOffset - SMALLSNIPPETLENGTH);
 
-            region.CharLength = originalRegion.CharLength + region.CharOffset + smallSnippetLength < newLineIndex.Text.Length
-                ? originalRegion.CharLength + smallSnippetLength + Math.Abs(region.CharOffset - originalRegion.CharOffset)
-                : newLineIndex.Text.Length - region.CharOffset;
+            region.CharLength =
```

Comment:
>CharLength [](http://example.com/codeflow?start=19&length=10)

Similarly, any context region that blows past the end of the actual file is invalid.